### DISCUSSION TITLE
Multi-instance folders are not side-by-side

### DISCUSSION THREAD

**Discussion by ehavelaarsnz** [wixtoolset/issues#7999] [Questions] [answered]
How do I get multiple instances to install in folders side-by-side?  If I do the following, they all target folder with the default ProductName instead of the one specified in the instance.

``` wxs
    <InstanceTransforms Property="MY_ENVIRONMENT">
      <Instance Id="TST" ProductCode="..." ProductName="MyProduct TEST"/>
      <Instance Id="UAT" ProductCode="..." ProductName="MyProduct UAT"/>
    </InstanceTransforms>

    <StandardDirectory Id="ProgramFiles6432Folder">
      <Directory Id="INSTALLFOLDER" Name="!(bind.property.Manufacturer)\!(bind.property.ProductName)" />
    </StandardDirectory>
```

**Comment by chrpai** [comment, marked as answer]
Binding is a build time thing.  For a runtime thing you need to call set directory or setpropery to mutate the path.

Honestly, I mastered multiple instance installations about 18 years ago and used them until about 12 years ago.  At some point I realized that they are just kind of more pain then they are worth.  Hardly anyone understands them.  If you only have a handful of instances like TEST and UAT then I'd just build 3 different packages that are a veneer that mutates the needed stuff and shares most of the fragments behind the scenes.

Then again I'd also ask if you really want DEV, TEST, QA, UAT and/or PROD on the same machine of it it would just make sense to spin up a few more VMs.

**Reply by ehavelaarsnz** [reply]
Since this is for a desktop app for a few hundred users who might want to use one of two instances, doubling the number of VM's is not really an option.

I've ended up using `<SetDirectory />`.  I know I can do multiple builds, but that just seemed... wrong.
